addons/analytic/tests/test_plan_operations.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+import psycopg2
+
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+from odoo.tools import mute_logger
+
+
+@tagged('post_install', '-at_install')
+class TestAnalyticPlanOperations(TransactionCase):
+    def test_delete_plan(self):
+        plan = self.env['account.analytic.plan'].create({'name': 'Test Plan'})
+        column = plan._column_name()
+
+        # columns exists
+        self.env.cr.execute(f"SELECT {column} FROM account_analytic_line LIMIT 1")
+
+        plan.unlink()
+        with self.assertRaises(psycopg2.errors.UndefinedColumn), mute_logger('odoo.sql_db'):
+            # column has been deleted
+            self.env.cr.execute(f"SELECT {column} FROM account_analytic_line LIMIT 1")
+
+    def test_delete_plan_with_view(self):
+        plan = self.env['account.analytic.plan'].create({'name': 'Test Plan'})
+        column = plan._column_name()
+
+        self.env['ir.ui.view'].create({
+            'name': 'Manual view',
+            'model': 'account.analytic.line',
+            'type': 'search',
+            'arch': f'<search><field name="{column}"/></search>',
+        })
+
+        # can't delete a plan still used in a view
+        with self.assertRaisesRegex(UserError, 'still present in views'):
+            plan.unlink()
+
+    def test_validate_deleted_account(self):
+        plan, mandatory_plan = self.env['account.analytic.plan'].create([{
+            'name': 'Test Plan',
+        }, {
+            'name': 'Mandatory Plan',
+            'default_applicability': 'mandatory',
+        }])
+        test_account, mandatory_account = self.env['account.analytic.account'].create([{
+            'name': 'Test Account',
+            'code': 'TAC',
+            'plan_id': plan.id,
+        }, {
+            'name': 'Mandatory Account',
+            'code': 'manda',
+            'plan_id': mandatory_plan.id,
+        }])
+        distribution = self.env['account.analytic.distribution.model'].create({}).with_context(validate_analytic=True)
```

Comment:
I would say that the variable name is not quite right (especially when you see the next line